Mathlib/Algebra/Category/CommAlgCat/Monoidal.lean

```diff
@@ -0,0 +1,126 @@
+/-
+Copyright (c) 2025 Yaël Dillies, Christian Merten, Michał Mrugała, Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Christian Merten, Michał Mrugała, Andrew Yang
+-/
+import Mathlib.Algebra.Category.CommAlgCat.Basic
+import Mathlib.CategoryTheory.Monoidal.Cartesian.Basic
+
+/-!
+# The cocartesian monoidal category structure on commutative `R`-algebras
+
+This file provides the cocartesian-monoidal category structure on `CommAlgCat R` constructed
+explicitly using the tensor product.
+-/
+
+open CategoryTheory CartesianMonoidalCategory Limits TensorProduct
+
+noncomputable section
+
+namespace CommAlgCat
+universe u v
+variable {R : Type u} [CommRing R] {A B : CommAlgCat.{u} R}
+
+variable (A B) in
+/-- The explicit cocone with tensor products as the fibered product in `CommAlgCat`. -/
+def binaryCofan : BinaryCofan A B :=
+  .mk (ofHom Algebra.TensorProduct.includeLeft)
+    (ofHom (Algebra.TensorProduct.includeRight (A := A)))
+
+variable (A B) in
+@[simp]
+lemma binaryCofan_inl : (binaryCofan A B).inl = ofHom Algebra.TensorProduct.includeLeft := rfl
+
+variable (A B) in
+@[simp]
+lemma binaryCofan_inr : (binaryCofan A B).inr = ofHom Algebra.TensorProduct.includeRight := rfl
+
+variable (A B) in
+@[simp] lemma binaryCofan_pt : (binaryCofan A B).pt = .of R (A ⊗[R] B) := rfl
+
+variable (A B) in
+/-- Verify that the pushout cocone is indeed the colimit. -/
+def binaryCofanIsColimit : IsColimit (binaryCofan A B) :=
+  BinaryCofan.IsColimit.mk _
+    (fun f g ↦ ofHom (Algebra.TensorProduct.lift f.hom g.hom fun _ _ ↦ .all _ _))
+    (fun f g ↦ by ext1; exact Algebra.TensorProduct.lift_comp_includeLeft _ _ fun _ _ ↦ .all _ _)
+    (fun f g ↦ by ext1; exact Algebra.TensorProduct.lift_comp_includeRight _ _ fun _ _ ↦ .all _ _)
+    (fun f g m hm₁ hm₂ ↦ by
+      ext1
+      refine Algebra.TensorProduct.liftEquiv.symm_apply_eq (y := ⟨⟨_, _⟩, fun _ _ ↦ .all _ _⟩).mp ?_
+      exact Subtype.ext (Prod.ext congr(($hm₁).hom) congr(($hm₂).hom)))
+
+/-- The initial object of `CommAlgCat R` is `R` as an algebra over itself -/
+def isInitialSelf : IsInitial (of R R) := .ofUniqueHom (fun A ↦ ofHom (Algebra.ofId R A))
+  fun _ _ ↦ hom_ext (Algebra.ext_id _ _ _)
+
+open Opposite Algebra.TensorProduct CommAlgCat MonoidalCategory
+
+attribute [local simp] Algebra.TensorProduct.one_def in
+instance : MonoidalCategory (CommAlgCat.{u} R) where
+  tensorObj S T := of R (S ⊗[R] T)
+  whiskerLeft _ {_ _} f := ofHom (Algebra.TensorProduct.map (.id _ _) f.hom)
+  whiskerRight {_ _} f T := ofHom (Algebra.TensorProduct.map f.hom (.id _ _))
+  tensorHom {_ _ _ _} f g := ofHom (map f.hom g.hom)
+  tensorUnit := .of R R
+  associator {_ _ _} := isoMk (Algebra.TensorProduct.assoc R R _ _ _)
+  leftUnitor _ := isoMk (Algebra.TensorProduct.lid R _)
+  rightUnitor _ := isoMk (Algebra.TensorProduct.rid R R _)
+
+@[simp] lemma coe_tensorUnit : 𝟙_ (CommAlgCat.{u} R) = R := rfl
```

Comment:
```suggestion
@[simp] lemma coe_tensorUnit : 𝟙_ (CommAlgCat R) = R := rfl

section
variable (A B : CommAlgCat R)

```

Removing the universe parameter outside of the instance definitions seems to work. Was there a reason it was made explicit everywhere?